zerver/views/registration.py

```diff
@@ -243,6 +243,12 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
         timezone = ""
         if 'timezone' in request.POST and request.POST['timezone'] in get_all_timezones():
             timezone = request.POST['timezone']
+        twenty_four_hour_time = realm.default_twenty_four_hour_time
+        if 'time_format' in request.POST:
+            if request.POST['time_format'] == "12":
+                twenty_four_hour_time = False
+            elif request.POST['time_format'] == "24":
+                twenty_four_hour_time = True
```

Comment:
Regarding boolean fields, I think it was this part in the form @showell and I were talking about making a boolean. The field currently expects a string `time_format` which we could just pass boolean instead of passing `"12"` and `"24"` (similar to what is there in the backend).

Reply:
ok